src/AuthorizationUtils.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+/*
+ * This file is part of the Mercure Component project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Symfony\Bundle\MercureBundle;
+
+use Symfony\Component\HttpFoundation\Cookie;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\Mercure\Exception\InvalidArgumentException;
+use Symfony\Component\Mercure\Hub;
+use Symfony\Component\Mercure\Jwt\TokenFactoryInterface;
+use Symfony\Component\WebLink\GenericLinkProvider;
+use Symfony\Component\WebLink\Link;
+
+final class AuthorizationUtils
+{
+    private const MERCURE_AUTHORIZATION_COOKIE_NAME = 'mercureAuthorization';
+
+    private $hubs;
+    private $factories;
+
+    /**
+     * @param iterable<string, Hub>                   $hubs
+     * @param iterable<string, TokenFactoryInterface> $factories
+     */
+    public function __construct(iterable $hubs, iterable $factories)
+    {
+        $this->hubs = \is_array($hubs) ? $hubs : iterator_to_array($hubs);
+        $this->factories = \is_array($factories) ? $factories : iterator_to_array($factories);
+    }
+
+    /**
+     * Add mercure link header to the given request.
+     */
+    public function addLink(string $hub, Request $request): void
```

Comment:
This isn't related to authorization but to discovery. Maybe can we find a better name for this class?

Reply:
Also, `$hub` should be optional, and the default hub should be used by default (same for `createCookie()`). This will improve the DX.

Reply:
we can/probably should split this into two.

Reply:
done 👍